client/src/components/Workflow/Editor/Node.vue

```diff
@@ -175,7 +175,9 @@ const workflowOutputs = computed(() => props.step.workflow_outputs || []);
 const connectionStore = useConnectionStore();
 const stateStore = useWorkflowStateStore();
 const stepStore = useWorkflowStepStore();
-const isLoading = computed(() => Boolean(stateStore.getStepLoadingState(props.id)?.loading));
+const isLoading = computed(() =>
+    Boolean(stateStore.getStepLoadingState(props.id)?.loading && props.step.outputs.length === 0)
```

Comment:
```suggestion
    Boolean(stateStore.getStepLoadingState(props.id)?.loading)
```
happy to merge that then

Reply:
I was about to suggest this.